### PR TITLE
Add `version.txt` into build to prevent cache migrations

### PR DESCRIPTION
**Pull request by varunshenoy**
When we download files using `huggingface_hub`, we don't include a `version.txt` in the cache directory set to 1. If this file isn't present, `transformers` will run a cache migration (which is basically a no-op).

This PR adds a `version.txt` into the build so that we can avoid any unnecessary cache migrations.

### CODE REVIEW

File: truss/templates/server.Dockerfile.jinja [hunk 1]
```diff
@@ -46,12 +46,17 @@ RUN pip install -r server_requirements.txt --no-cache-dir && rm -rf /root/.cache
 {% endblock %}
 
 {% block cache_weights %}
+ARG HF_SECRET_TOKEN
```
Comment (bolasim) [inline truss/templates/server.Dockerfile.jinja]: I think you might wanna drop this arg stuff for the next PR

File: truss/templates/server.Dockerfile.jinja [hunk 2]
```diff
@@ -64,7 +69,9 @@ RUN $PYTHON_EXECUTABLE /cache_warmer.py {{file}} {{repo}}
 {% block app_copy %}
 # Copy data before code for better caching
 {%- if data_dir_exists %}
-COPY ./{{config.data_dir}} /app/data
+    {%- for file_list in data_files%}
```
Comment (bolasim) [inline truss/templates/server.Dockerfile.jinja]: this seem unerelated too

**Review by bolasim — COMMENTED** [review]
What happens if the `transformers` library is a different version and has a different version.txt?

**Review by bolasim — APPROVED** [review]

**Comment by joostinyi** [comment]
Can you point to some documentation/your findings that lead you to this fix and add a comment to the changed file?

**Comment by varunshenoy** [comment]
> Can you point to some documentation/your findings that lead you to this fix and add a comment to the changed file?

https://github.com/huggingface/transformers/blob/0fd8d2aa2cc9e172a8af9af8508b2530f55ca14c/src/transformers/utils/hub.py#L1180

**Pull request merged**